SConstruct

```diff
@@ -251,6 +251,19 @@ else:
         print("Automatically detected platform: " + selected_platform)
         env_base["platform"] = selected_platform
 
+if selected_platform == "x11":
+    # Alias `x11` to `linuxbsd` for compatibility.
+    # This way, outdated documentation will still be usable to produce a working binary.
+    print('Platform "x11" has been renamed to "linuxbsd" in Godot 4.0. '
+          'Building for platform "linuxbsd".')
+    selected_platform = "linuxbsd"
+
+if selected_platform == "linux" or selected_platform == "bsd":
+    # Alias `linux` and `bsd` to `linuxbsd` for convenience.
+    print('Platform "%s" is an alias for "linuxbsd". ' % selected_platform +
+          'Building for platform "linuxbsd".')
```

Comment:
I wouldn't make it print anything, that's too much information. Just like `p` is a silent alias for `platform`, `bsd` and `linux` can be silent aliases for `linuxbsd`.